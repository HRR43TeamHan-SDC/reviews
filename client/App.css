@font-face {
  font-family: 'BrandonText';
  src:  url('./BrandonText-Regular.otf') format('opentype');
}

body {
  font-family: 'BrandonText';
}

.review {
  display: flex;
  flex-direction: row;
  width: 100%;
  padding-top: 1rem;
  padding-bottom: 1rem;
  border-bottom: solid 1px #d8d9db;
  color: #2d333f;
}

.review-user {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-right: 1rem;
  margin-left: 1rem;
  width: 6rem;
  position: relative;
  flex: none;
}

.user-box {
  display: flex;
  justify-content: center;
  margin-bottom: 0.5rem;
}

.user-circle {
  background: #6c8ae4;
  width: 3rem;
  height: 3rem;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.user-initials {
  color: #ffffff;
  font-weight: 500;
  text-transform: capitalize;
}

.user-name {
  font-weight: 500;
  font-size: 0.875rem;
  margin-bottom: 0.25rem;
  text-align: center;
  word-break: break-word;
  color: #2d333f;
}

.user-vip {
  text-transform: uppercase;
  position: absolute;
  background-color: #fdaf08;
  padding: 2px 0.5rem;
  font-size: 12px;
  border-radius: 1rem;
  font-weight: 700;
  top: -2px;
  color: #fff;
  left: 0;
}

.user-city {
  font-size: 0.875rem;
  font-weight: 500;
  text-align: center;
  line-height: 1.43;
  color: #6f737b;
  margin-bottom: 0.5rem;
}

.user-num-reviews {
  color: #6f737b;
  display: flex;
  flex-direction: row;
  font-size: 12px;
  font-weight: bold;
  line-height: 1.43;
  flex: none;
  white-space: nowrap;
}

.msg-icon {
  /* background-image: url('./message-icon.png'); */
  display: inline-block;
  position: relative;
  height: 12px;
  width: 12px;
  flex: none;
  margin-right: 0.25rem;
  margin-top: 0.15rem;
}

.review-details {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.review-header {
  display: flex;
  position: relative;
  flex-direction: column;
}

.review-pre-ratings {
  display: flex;
  justify-content: flex-start;
  align-items: center;
}

.review-stars {
  font-size: 1.5rem;
  font-weight: 500;
  color: #d8d9db;
}

.review-date {
  font-size: 0.875rem;
  line-height: 1.43;
  margin: 0 0.25rem;
}

.review-ratings {
  /* display: flex;
  flex-direction: row; */
  max-width: 992px;
  margin-top: 0.5rem;
  font-size: 0.875rem;
  font-weight: bold;
  line-height: 1.43;
}

.review-numbers {
  color: #da3743;
}

.review-comments {
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  line-height: 1.5;
  font-size: 1rem;
  font-weight: normal;
}