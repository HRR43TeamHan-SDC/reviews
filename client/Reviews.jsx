import React, { useState, useEffect } from 'react';
import msgIcon from './message-icon.png';

const moment = require('moment');

const Reviews = ({ reviews }) => {
  const list = reviews.map((review) => {
    return (
      <div key={review._id} className="review">
        <div className="review-user">
          <div className="user-box">
            <div className="user-circle">
              <div className="user-initials">
                {review.firstName[0]}{review.lastName[0]}
              </div>
            </div>
          </div>
          <div className="user-name">
            {review.firstName}
            {review.lastName.substring(0, 1)}
            {review.vip ? <span className="user-vip">VIP</span> : ''}
          </div>
          <div className="user-city">
            {review.city}
          </div>
          <div className="user-num-reviews">
            <img className="msg-icon" src={msgIcon}></img>
            {review.numReviews}
            {' '}
            reviews
          </div>
        </div>
        <div className="review-details">
          <div className="review-header">
            <div className="review-pre-ratings">
              <div className="review-stars">
                <span style={{ color: (review.overall >= 1) ? 'rgb(218, 55, 67)' : '#d8d9db' }}>★</span>
                <span style={{ color: (review.overall >= 2) ? 'rgb(218, 55, 67)' : '#d8d9db' }}>★</span>
                <span style={{ color: (review.overall >= 3) ? 'rgb(218, 55, 67)' : '#d8d9db' }}>★</span>
                <span style={{ color: (review.overall >= 4) ? 'rgb(218, 55, 67)' : '#d8d9db' }}>★</span>
                <span style={{ color: (review.overall >= 5) ? 'rgb(218, 55, 67)' : '#d8d9db' }}>★</span>
              </div>
              {' '}
              <span className="review-date">
                • Dined {moment(review.dineDate).fromNow()}
              </span>
            </div>
              <div className="review-ratings">
                Overall
                {' '}
                <span className="review-numbers">{review.overall}
                </span>
                {' '}
                • Food
                {' '}
                <span className="review-numbers">{review.food}</span>
                {' '}
                • Service
                {' '}
                <span className="review-numbers">{review.service}</span>
                {' '}
                • Ambience
                {' '}
                <span className="review-numbers">{review.ambience}</span>

              </div>
          </div>
          <div className="review-comments">
            {review.comments}
          </div>
          <br></br>
        </div>

      </div>

    );
  });
  return <div>{list}</div>;
};
export default Reviews;
