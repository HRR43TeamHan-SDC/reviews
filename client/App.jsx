import React, { useState, useEffect } from 'react';
// import React from 'react';
import axios from 'axios';
import Reviews from './Reviews.jsx';

const App = () => {
  const [reviews, setReviews] = useState([]);

  useEffect(() => {
    const queryString = window.location.search;
    const urlParams = new URLSearchParams(queryString);
    const id = urlParams.get('id');
    axios.get(`/${id}`)
      .then((result) => {
        console.log('data from DB ', result.data);
        setReviews(result.data);
        console.log('data in state ', reviews);
      })
      .catch((err) => console.log('error ', err));
  }, []);

  return (
    <Reviews reviews={reviews} />
  );
};

// const App = () => <div>testing React!</div>;
export default App;
