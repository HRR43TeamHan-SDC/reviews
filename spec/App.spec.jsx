import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import App from '../client/App.jsx';
import Reviews from '../client/Reviews.jsx';

describe('App', () => {
  test('App renders correctly with a snapshot', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('App renders Reviews', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(Reviews).length).toEqual(1);
  });
});
