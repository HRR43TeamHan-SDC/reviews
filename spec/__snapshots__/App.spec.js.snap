// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders correctly with a snapshot 1`] = `<div />`;
